### PR TITLE
Throw NotSupportedException when stream length is zero.

### LINKED ISSUE #1903: Image.Identify throws ArgumentOutOfRangeException on empty unseekable stream
### Prerequisites

- [x] I have written a descriptive issue title
- [x] I have verified that I am running the latest version of ImageSharp
- [ ] I have verified if the problem exist in both `DEBUG` and `RELEASE` mode
- [x] I have searched [open](https://github.com/SixLabors/ImageSharp/issues) and [closed](https://github.com/SixLabors/ImageSharp/issues?q=is%3Aissue+is%3Aclosed) issues to ensure it has not already been reported

### Description
`Image.Identify` throws `ArgumentOutOfRangeException` when provided an empty unseekable stream.

Stack trace:
```
System.ArgumentOutOfRangeException: Specified argument was out of the range of valid values.
Parameter name: value
   at SixLabors.ImageSharp.IO.ChunkedMemoryStream.ThrowArgumentOutOfRange(String value)
   at SixLabors.ImageSharp.IO.ChunkedMemoryStream.set_Position(Int64 value)
   at SixLabors.ImageSharp.Image.WithSeekableStream[T](Configuration configuration, Stream stream, Func`2 action)
   at SixLabors.ImageSharp.Image.Identify(Configuration configuration, Stream stream, IImageFormat& format)
   at SixLabors.ImageSharp.Image.Identify(Stream stream)
```

### Steps to Reproduce
```C#
using SixLabors.ImageSharp;
using System.IO;
using System.IO.Compression;

internal class Test
{
    public static void Main()
    {
        using (var zipFile = new ZipArchive(new MemoryStream(new byte[] {
            0x50, 0x4B, 0x03, 0x04, 0x0A, 0x00, 0x00, 0x00, 0x00, 0x00, 0x77, 0xAF,
            0x94, 0x53, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00,
            0x00, 0x00, 0x07, 0x00, 0x00, 0x00, 0x6D, 0x79, 0x73, 0x74, 0x65, 0x72,
            0x79, 0x50, 0x4B, 0x01, 0x02, 0x3F, 0x00, 0x0A, 0x00, 0x00, 0x00, 0x00,
            0x00, 0x77, 0xAF, 0x94, 0x53, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00,
            0x00, 0x00, 0x00, 0x00, 0x00, 0x07, 0x00, 0x24, 0x00, 0x00, 0x00, 0x00,
            0x00, 0x00, 0x00, 0x20, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x6D,
            0x79, 0x73, 0x74, 0x65, 0x72, 0x79, 0x0A, 0x00, 0x20, 0x00, 0x00, 0x00,
            0x00, 0x00, 0x01, 0x00, 0x18, 0x00, 0x46, 0x82, 0xFF, 0x91, 0x27, 0xF6,
            0xD7, 0x01, 0x55, 0xA1, 0xF9, 0x91, 0x27, 0xF6, 0xD7, 0x01, 0x55, 0xA1,
            0xF9, 0x91, 0x27, 0xF6, 0xD7, 0x01, 0x50, 0x4B, 0x05, 0x06, 0x00, 0x00,
            0x00, 0x00, 0x01, 0x00, 0x01, 0x00, 0x59, 0x00, 0x00, 0x00, 0x25, 0x00,
            0x00, 0x00, 0x00, 0x00 })))
        using (var stream = zipFile.Entries[0].Open())
            Image.Identify(stream);
    }
}
```

### System Configuration

- ImageSharp version: 2.0.0-alpha.0.140
- Other ImageSharp packages and versions: N/A
- Environment (Operating system, version and so on): Windows 10
- .NET Framework version: 4.8.1
- Additional information: N/A

### PR DESCRIPTION
**Pull request by JimBobSquarePants**
### Prerequisites

- [x] I have written a descriptive pull-request title
- [x] I have verified that there are no overlapping [pull-requests](https://github.com/SixLabors/ImageSharp/pulls) open
- [x] I have verified that I am following the existing coding patterns and practice as demonstrated in the repository. These follow strict Stylecop rules :cop:.
- [x] I have provided test coverage for my change (where applicable)

### Description

 Fixes  #1903

We should treat streams of length zero in the same manner as non readable streams by throwing a  `NotSupportedException`.

### CODE REVIEW

File: src/ImageSharp/Image.FromStream.cs [hunk 1]
```diff
@@ -724,7 +724,7 @@ public static Image Load(Configuration configuration, Stream stream, out IImageF
             Guard.NotNull(configuration, nameof(configuration));
             Guard.NotNull(stream, nameof(stream));
 
-            if (!stream.CanRead)
+            if (!stream.CanRead || stream.Length == 0)
```
Comment (dlemstra) [inline src/ImageSharp/Image.FromStream.cs:727]: Not sure if you should do this `.Length` check. Don't thinks this will work for non seekable streams.

Comment (JimBobSquarePants) [reply src/ImageSharp/Image.FromStream.cs:727]: TIL. Will update. https://docs.microsoft.com/en-us/dotnet/api/system.io.stream.length?redirectedfrom=MSDN&view=net-6.0#exceptions

Comment (JimBobSquarePants) [reply src/ImageSharp/Image.FromStream.cs]: Soooo. It turns out those docs are unreliable. [SubReadStream](https://source.dot.net/#System.IO.Compression/System/IO/Compression/ZipCustomStreams.cs,246) is non-seekable but supports the Length property which makes writing a condition impossible.

**Comment by JimBobSquarePants** [comment]
Closed in favor of #1907

**Pull request closed without merging**